### PR TITLE
Add CMake Presets to the CMake Integration docs

### PR DESCRIPTION
**Pull request by ras0219-msft**

### CODE REVIEW

File: vcpkg/users/buildsystems/cmake-integration.md [hunk 1]
```diff
@@ -30,25 +52,13 @@ include_directories(${CATCH_INCLUDE_DIR})
...
-```json
-{
-  "cmake.configureSettings": {
-    "CMAKE_TOOLCHAIN_FILE": "<vcpkg-root>/scripts/buildsystems/vcpkg.cmake"
-  }
-}
-```
-
-### Visual Studio
+Learn more about Visual Studio at [Configure and build with CMake Presets in Visual Studio](/cpp/build/cmake-presets-vs).
```
Comment (AugP) [inline vcpkg/users/buildsystems/cmake-integration.md:59]: Suggested reword: "Learn more at: <LINK>". No point saying Visual Studio twice when it's in the URL anyway.

File: vcpkg/users/buildsystems/cmake-integration.md [hunk 2]
```diff
@@ -30,25 +52,13 @@ include_directories(${CATCH_INCLUDE_DIR})
...
 
-In the CMake Settings Editor, add the path to the vcpkg toolchain file under `CMake toolchain file`:
-
-```cmake
-<vcpkg-root>/scripts/buildsystems/vcpkg.cmake
-```
+Learn more about Visual Studio Code at [Configure and build with CMake Presets in Visual Studio Code](https://github.com/microsoft/vscode-cmake-tools/blob/main/docs/cmake-presets.md).
```
Comment (AugP) [inline vcpkg/users/buildsystems/cmake-integration.md:61]: Same feedback as above, just reword to "Learn more at <LINK>"

**Review by AugP — APPROVED** [review]
LGTM, left 2 minor comments, looks good to merge overall.

File: vcpkg/users/buildsystems/cmake-integration.md [hunk 3]
```diff
@@ -12,13 +12,35 @@ See [Installing and Using Packages Example: sqlite](../../examples/installing-an
...
+
+```sh
+cmake -B build -S /my/project --preset debug
+```
+
+If you need to use an absolute path for vcpkg specific to your current machine, you can use `CMakeUserPresets.json` and add it to your `.gitignore` file.
+
+:::code language="json" source="snippets/cmake-integration/CMakeUserPresets.json":::
+
+CMake versions older than 3.19 must pass the toolchain file on the configure command line:
```
Comment (Osyotr) [inline vcpkg/users/buildsystems/cmake-integration.md:27]: Or use `cacheVariables` map

Comment (ras0219-msft) [reply vcpkg/users/buildsystems/cmake-integration.md:27]: CMakePresets support was introduced in 3.19, so older than that must pass it on the command line :). Technically, 3.19 and 3.20 can pass it via `cacheVariables` but not `toolchainFile`. I didn't feel it was helpful to exhaustively enumerate all the possibilities (the `CMAKE_TOOLCHAIN_FILE` environment variable, for example) -- so I stuck to "Recent users should do X, older users should do Y".

**Pull request merged**